wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessConnectionPlugin.java

```diff
@@ -224,21 +225,24 @@ private Connection retryConnection(final List<HostSpec> limitlessRouters, final
     return retryConnection(newLimitlessRouters, props, retryCount, originalException);
   }
 
-  private synchronized List<HostSpec> synchronouslyGetLimitlessRouters(final Properties props)
+  private List<HostSpec> synchronouslyGetLimitlessRouters(final Properties props)
       throws SQLException {
-    final List<HostSpec> currentLimitlessRouters = this.limitlessRouterService.getLimitlessRouters(
-        this.pluginService.getHostListProvider().getClusterId(), props);
-    if (!currentLimitlessRouters.isEmpty()) {
-      return currentLimitlessRouters;
-    }
+    synchronized (synchronouslyGetLimitlessRoutersLock) {
+      final List<HostSpec> currentLimitlessRouters = this.limitlessRouterService.getLimitlessRouters(
+          this.pluginService.getHostListProvider().getClusterId(), props);
+      if (!currentLimitlessRouters.isEmpty()) {
+        return currentLimitlessRouters;
+      }
 
-    LOGGER.fine(Messages.get("LimitlessConnectionPlugin.synchronouslyGetLimitlessRouters"));
-    List<HostSpec> newLimitlessRouters = this.limitlessRouterService.forceGetLimitlessRouters(
-        this.pluginService.getHostListProvider().getClusterId(), props);
-    if (newLimitlessRouters != null && !newLimitlessRouters.isEmpty()) {
-      return newLimitlessRouters;
+      LOGGER.fine(Messages.get("LimitlessConnectionPlugin.synchronouslyGetLimitlessRouters"));
+      List<HostSpec> newLimitlessRouters = this.limitlessRouterService.forceGetLimitlessRouters(
+          this.pluginService.getHostListProvider().getClusterId(), props);
+      if (newLimitlessRouters != null && !newLimitlessRouters.isEmpty()) {
+        return newLimitlessRouters;
+      }
     }
 
+    List<HostSpec> newLimitlessRouters = Collections.emptyList();
```

Comment:
This is already declared above L238. Does Java not have a compile error when this happens?

Reply:
Ah, nevermind. Scoped differently. Hard to see on Github